File: skills/autospec-shared/tests/fixtures/tree-sitter/typescript/sample.ts
// sample.ts — TypeScript fixture for autospec-docs walker unit tests.

import { readFileSync } from 'fs';
import path from 'path';

export interface Config {
    host: string;
    port: number;
}

export type StatusCode = 200 | 400 | 404 | 500;

export const DEFAULT_PORT = 3000;

export function parseConfig(filePath: string): Config {
    const raw = readFileSync(filePath, 'utf8');
    return JSON.parse(raw) as Config;
}

export class Server {
    private config: Config;
    constructor(config: Config) {
        this.config = config;
    }
    start(): void {
        console.log(`Listening on ${this.config.host}:${this.config.port}`);
    }
}

File: skills/autospec-shared/tests/fixtures/tree-sitter/python/sample.py
"""sample.py — Python fixture for autospec-docs walker unit tests."""

import os
import json
from pathlib import Path

MAX_RETRIES = 3
DEFAULT_TIMEOUT = 30


def parse_config(file_path: str) -> dict:
    """Parse a JSON config file and return the result."""
    with open(file_path) as f:
        return json.load(f)


def format_message(template: str, **kwargs) -> str:
    """Format a message template with keyword arguments."""
    return template.format(**kwargs)


class ConfigLoader:
    """Loads and caches configuration from disk."""

    def __init__(self, base_dir: str):
        self.base_dir = Path(base_dir)
        self._cache: dict = {}

    def load(self, name: str) -> dict:
        if name not in self._cache:
            self._cache[name] = parse_config(str(self.base_dir / f"{name}.json"))
        return self._cache[name]


if __name__ == "__main__":
    loader = ConfigLoader(os.getcwd())
    print(loader.load("default"))
